### ISSUE TITLE
Dialogue file index.php gives 3 debug messages - one is FATAL

### ISSUE THREAD

**Issue by drachels** [troywilliams/moodle-mod_dialogue#40]
I'm using the latest version of Dialogue. I just clicked on the Dialogue link in an Activities block of a course and got a deprecated, add to log, and fatal debug message as follows:

get_context_instance() is deprecated, please use context_xxxx::instance() instead.

```
line 256 of /lib/deprecatedlib.php: call to debugging()
line 33 of /mod/dialogue/index.php: call to get_context_instance()
```

add_to_log() has been deprecated, please rewrite your code to the new events API

```
line 50 of /lib/deprecatedlib.php: call to debugging()
line 35 of /mod/dialogue/index.php: call to add_to_log()
```

Fatal error: Class 'dialogue' not found in /var/www/moodlexx/moodle/mod/dialogue/index.php on line 58

Call Stack:
0.0001 232312 1. {main}() /var/www/moodlexx/moodle/mod/dialogue/index.php:0

**Comment by drachels** [comment]
A possible fix for the get context instance message is to change line 33 to 
$coursecontext = context_course::instance($course->id);

A possible fix for the add to log is
// Trigger course module instance list event.
$params = array(
    'context' => context_course::instance($course->id)
);
$event = \mod_dialogue\event\course_module_instance_list_viewed::create($params);
$event->add_record_snapshot('course', $course);
$event->trigger();

**Comment by drachels** [comment]
I haven't come up with anything that works to fix the fatal error.

**Comment by gibbisons** [comment]
Please advise which version of Moodle you are using.

**Comment by drachels** [comment]
At the time I posted those two fixes it was in Moodle 3.0, but I have implemented the same changes into my Moodle 3.1 site as well.

**Comment by choggnz** [comment]
I've found a fix for the fatal error by prefixing the dialogue class with its namespace. So on line 61 of index.php: 

array(dialogue::STATE_OPEN, dialogue::STATE_CLOSED) becomes: 

array(mod_dialogue\dialogue::STATE_OPEN, mod_dialogue\dialogue::STATE_CLOSED)

I've only tested this on Moodle 3.1

**Comment by drachels** [comment]
Thanks for the info. Your fix works in Moodle 3.0 also. At first, I missed that it is actually two changes in that line of code.

**Comment by troywilliams** [comment]
Closing, has been fixed. Thanks for your contribution to making dialogue better 👍
